nomad/state/state_store.go

```diff
@@ -1828,6 +1839,15 @@ func (s *StateStore) CSIVolumeDeregister(index uint64, ids []string) error {
 			return fmt.Errorf("volume not found: %s", id)
 		}
 
+		vol, ok := existing.(*structs.CSIVolume)
+		if !ok {
+			return fmt.Errorf("volume not found: %s", id)
+		}
+
+		if namespace != vol.Namespace {
```

Comment:
Seems like we're missing some code in the body here?

Reply:
Ugh, thanks. This was the snippet of code from the ACL branch that got me started.